### ISSUE TITLE
dart2js: investigate: lazy-initialization of large constants

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#32943]
It might be cheaper to initialize large constants more like a lazy static initializer instead of precomputing the value during startup. This would depend on the startup cost of evaluating the expression vs the cost of creating a function that evaluates the expression.

#25230 would be useful in aggregating the non-shared sub-constants of the constant into a larger expression which might be worth deferring even though the non-aggregated parts are not. 

Lazy-init constants could reference other lazy-init constants, but eager-initialized constants should not, since that would force the lazy-init to be executed early.
Since constants cannot fail, the full mechanics of lazy static initialization are not required.

**Comment by srawlins** [comment]
@rakudrama Is this still possible / relevant with const-update-2018?

**Comment by sigmundch** [comment]
Yes. In essence, the way we represent constants in the generated code is the same as it was before const-update-2018 change. This issue is mostly about how we reduce the load-time cost of initializing constant values in the generated code.
